### ISSUE TITLE
avoid turning `\n` into `\\n` in roxygen docs

### ISSUE THREAD

**Issue by IndrajeetPatil** [r-lib/styler#728]
As discussed [here](https://github.com/easystats/insight/pull/297#issuecomment-773340800):

As you mentioned, the issue is `\dontrun`:

```r
#' Data frame and Tables Pretty Formatting
#'
#' @examples
#' \dontrun{
#'   c("\nA yellow line", "yellow")
#' }
#' @export
NULL
```

**Comment by lorenzwalthert** [comment]
Thanks. The issue is actually really a really annoying if you want to get all cases right. I think we might eventually be able to leverage `{roxygen2}` infrastructure to *emulate* what the written Rd would look like from the roxygen comment (because they already spent like a reasonable amount of time on this, see screenshot -.-) and then continue to use `tools::tools::parse_Rd()`, which was written to parse escaped Rd, not roxygen comments. 
![Screenshot 2021-02-05 at 09 31 42](https://user-images.githubusercontent.com/10477073/107009175-f816da00-6794-11eb-9122-af73ba364724.png)

**Comment by lorenzwalthert** [comment]
@IndrajeetPatil as this is a very critical issue, I also aimed to push a fix quickly. Can you check if the associated PR works for you?

**Comment by IndrajeetPatil** [comment]
Just tried it and, yaay, that worked  🎉 

Thanks for solving this SO quickly!

**Comment by lorenzwalthert** [comment]
You probably won't believe it but it took me probably 8h to solve this annoying escape problem. And the best part: I did not even resolve it myself. It was mostly Hadley who implemented the Roxygen -> Rd translation that saved me. I can see far when I stand on the shoulders of giants.

**Comment by IndrajeetPatil** [comment]
I feel you, mate. Thanks a ton for all the hard work you have put into this amazing package. 

I use it on a daily basis, not only in my packages but also in **every** script I can get my hands on (mine or not)! 😅 
Can't imagine my workflow without it.

**Comment by lorenzwalthert** [comment]
Thanks for continuously filing issues when you find problems. I wouldn’t find all these bugs myself 🙈Little did I know that styler was going to see hundreds of (more) bugs at the time we released v1.0 “with confidence” in late 2017 🙀
